test/test_autograd.py

```diff
@@ -1786,6 +1796,20 @@ def test_set_grad_enabled(self):
             y = x * 2
         self.assertTrue(y.requires_grad)
 
+        @torch.set_grad_enabled(False)
+        def doubler_without(x):
+            return x * 2
+
+        y = doubler_without(x)
+        self.assertFalse(y.requires_grad)
+
+        @torch.set_grad_enabled(True)
+        def doubler_with(x):
+            return x * 2
+
+        y = doubler_with(x)
+        self.assertTrue(y.requires_grad)
```

Comment:
A better test would be to set grad mode to false, and only then call this function and do the assert. Right now it would pass if the function would be a no-op, because the grad mode is enabled in the env by default

Reply:
fixed!